FrameMetadata/Lets-Build-OTPublisher/TBExampleVideoCapture.m

```diff
@@ -0,0 +1,891 @@
+                                          planes:planes];
+    } else {
+        NSLog(@"No sanitization implementation for pixelFormat %d",
+              pixelFormat);
+        *data = NULL;
+        return 0;
+    }
+}
+
+- (size_t)sanitizeBiPlanarImageBuffer:(CVImageBufferRef)imageBuffer
+                                 data:(uint8_t**)data
+                               planes:(NSPointerArray*)planes
+{
+    size_t sanitaryBufferSize = 0;
+    for (int i = 0; i < CVPixelBufferGetPlaneCount(imageBuffer); i++) {
+        size_t planeImageWidth =
+        // TODO: (Apple bug?) biplanar pixel format reports 1/2 the width of
+        // what actually ends up in the pixel buffer for interleaved chroma.
+        // The only thing I could do about it is use image width for both plane
+        // calculations, in spite of this being technically wrong.
+        //CVPixelBufferGetWidthOfPlane(imageBuffer, i);
+        CVPixelBufferGetWidth(imageBuffer);
+        size_t planeImageHeight =
+        CVPixelBufferGetHeightOfPlane(imageBuffer, i);
+        sanitaryBufferSize += (planeImageWidth * planeImageHeight);
+    }
+    uint8_t* newImageBuffer = malloc(sanitaryBufferSize);
+    size_t bytesCopied = 0;
+    for (int i = 0; i < CVPixelBufferGetPlaneCount(imageBuffer); i++) {
+        [planes addPointer:&(newImageBuffer[bytesCopied])];
+        void* planeBaseAddress =
+        CVPixelBufferGetBaseAddressOfPlane(imageBuffer, i);
+        size_t planeDataWidth =
+        CVPixelBufferGetBytesPerRowOfPlane(imageBuffer, i);
+        size_t planeImageWidth =
+        // Same as above. Use full image width for both luma and interleaved
+        // chroma planes.
+        //CVPixelBufferGetWidthOfPlane(imageBuffer, i);
+        CVPixelBufferGetWidth(imageBuffer);
+        size_t planeImageHeight =
+        CVPixelBufferGetHeightOfPlane(imageBuffer, i);
+        for (int rowIndex = 0; rowIndex < planeImageHeight; rowIndex++) {
+            memcpy(&(newImageBuffer[bytesCopied]),
+                   &(planeBaseAddress[planeDataWidth * rowIndex]),
+                   planeImageWidth);
+            bytesCopied += planeImageWidth;
+        }
+    }
+    assert(bytesCopied == sanitaryBufferSize);
+    *data = newImageBuffer;
+    return bytesCopied;
+}
+
+- (void)captureOutput:(AVCaptureOutput *)captureOutput
+didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer
+       fromConnection:(AVCaptureConnection *)connection {
+    
+    if (!(_capturing && _videoCaptureConsumer)) {
+        return;
+    }
+    
+    // attach time stamp to a frame
+    
+    CMTime time = CMSampleBufferGetPresentationTimeStamp(sampleBuffer);
+    CVImageBufferRef imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+    
+    CVPixelBufferLockBaseAddress(imageBuffer, 0);
+    
+#warning custom method called here
+//    [self setCountToVideoFrame:_videoFrame];
+    
+    [self setTimestampToVideoFrame:_videoFrame];
+    
+//    [self setLocationToVideoFrame:_videoFrame];
```

Comment:
Just leave one thing here, and comment the rest of them in the README file instead.